app/jobs/decidim/spam_signal/spam_signal_job.rb

```diff
@@ -0,0 +1,26 @@
+
+
+
+module Decidim
+  module SpamSignal
+    class SpamSignalJob < ApplicationJob
+      
+      # attr_reader :logger
+      
+      def perform(logger = Rails.logger)
+        # @logger = logger
+        invoke!
+      end
+
+      private
+
+        def invoke!
+          puts "Init Job"
+          spam_signal = Decidim::SpamSignal::AppSpamSignal.new
```

Comment:
You can move the whole code of `Decidim::SpamSignal::AppSpamSignal` in the job. 
`.perform_later` goes in sidekiq, `perform_now` execute the code directly. No need for separation, we are on the same concern here.

Reply:
Sorry, I didn't understand, I can get off the code from lib and put it directly on the job?

Reply:
Yes, that what I meant but with the changes we discussed off-line I think most of this PR will be deprecated.